helm-charts/basehub/values.yaml

```diff
@@ -1447,17 +1455,32 @@ jupyterhub:
                   timeout=spawner.k8s_api_request_retry_timeout
               )
           c.Spawner.pre_spawn_hook = ensure_db_pvc
-      05-profile-groups: |
-        # Re-assignes c.KubeSpawner.profile_list to a callable that filters the
-        # initial configuration of profile_list based on the user's github
-        # org/team membership as declared via "allowed_groups" read from
-        # profile_list profiles.
+      05-auth-state-groups-key: |
+        # Pass the user's GitHub teams from auth_state as groups to the authenticator
+        async def custom_auth_state_groups_key(auth_state):
+          groups_list = []
+          if auth_state is None:
+            print("Auth state is None")
+            return groups_list
+          else:
+            for team in auth_state["teams"]:
+              if f'{team["organization"]["login"]}:{team["slug"]}' not in c.GitHubOAuthenticator.allowed_organizations:
+                continue
+              else:
+                groups_list.append(f'{team["organization"]["login"]}:{team["slug"]}')
+            return groups_list
+        if c.GitHubOAuthenticator:
+          c.GitHubOAuthenticator.allowed_organizations.append("2i2c-org:hub-access-for-2i2c-staff")
```

Comment:
This is unrelated, so it should be in its own separate entry rather than bundled in here.

Reply:
Now that we can set config for `GitHubOAuthenticator` in `basehub/values.yaml`, I have opted to include this in the list of `allowed_organizations` (see [Separate 2i2c-add-staff-gh-team access](https://github.com/2i2c-org/infrastructure/pull/6032/commits/43a7ae0a82fe5f10cc878ac7db18bb83aebed365))

Reply:
Actually, I have reverted that and separated this into its own extraConfig, since staff access needs to be *appended* to `allowed_organizations` specified in hub values files.